src/analysis/processing/qgsalgorithmmergevector.cpp

```diff
@@ -153,7 +158,7 @@ QVariantMap QgsMergeVectorAlgorithm::processAlgorithm( const QVariantMap &parame
           found = true;
           if ( destField.type() != sourceField.type() )
           {
-            feedback->pushInfo( QObject::tr( "%1 field in layer %2 has different data type than the destination layer (%3 instead of %4)" )
+            feedback->reportError( QObject::tr( "%1 field in layer %2 has different data type than the destination layer (%3 instead of %4)" )
```

Comment:
I would also add: *": Field converted to String type"* or something similar. ATM, the user is warned about the field dissimilarity but not about the action taken.

Reply:
I didn't run the alg but is it not what would be destField.typeName()?

Reply:
I didn't run either, but it seems that at this point it is still the original field type. It is converted to string in the next line `destField.setType( QVariant::String );`

Reply:
Ok. Before I push the various suggestions, would this (using %1 again) work?
```suggestion
            feedback->pushWarning( QObject::tr( "%1 field in layer %2 has different data type than the destination layer (%3 instead of %4). %1 field will be converted to string type." )
```

Reply:
Yes, both `%1` should be replaced by the first argument!